tests/beaker_tests/cisco_bgp/test_bgp.rb

```diff
@@ -0,0 +1,267 @@
+    'event_history_events'                   => 'size_small',
+    'event_history_periodic'                 => 'size_small',
+    'fast_external_fallover'                 => 'true',
+    'flush_routes'                           => 'false',
+    'graceful_restart'                       => 'true',
+    'graceful_restart_helper'                => 'false',
+    'graceful_restart_timers_restart'        => '120',
+    'graceful_restart_timers_stalepath_time' => '300',
+    'isolate'                                => 'false',
+    'log_neighbor_changes'                   => 'false',
+    'maxas_limit'                            => 'false',
+    'shutdown'                               => 'false',
+    'suppress_fib_pending'                   => 'false',
+    'timer_bestpath_limit'                   => '300',
+    'timer_bestpath_limit_always'            => 'false',
+    'timer_bgp_holdtime'                     => '180',
+    'timer_bgp_keepalive'                    => '60',
+  },
+}
+
+tests[:default_plat_1] = {
+  desc:           '1.2 Default Properties Platform-specific Part 1',
+  platform:       'n(3|9)k',
+  title_pattern:  "#{asn} red",
+  manifest_props: {
+    neighbor_down_fib_accelerate: 'default'
+  },
+  resource:       {
+    neighbor_down_fib_accelerate: 'false'
+  },
+}
+
+# Non-default Tests. NOTE: [:resource] = [:manifest_props] for all non-default
+
+tests[:non_def_1] = {
+  desc:           '2.1 Non Defaults Part 1',
+  title_pattern:  "#{asn} default",
+  manifest_props: {
+    # These properties are only configurable in default vrf
+    disable_policy_batching: 'true',
+    event_history_cli:       'size_medium',
+    event_history_detail:    'size_large',
+    event_history_events:    'size_disable',
+    event_history_periodic:  'false',
+  },
+}
+
+tests[:non_def_2] = {
+  desc:           '2.2 Non Defaults Part 2',
+  title_pattern:  "#{asn} default",
+  manifest_props: {
+    bestpath_always_compare_med:     'true',
+    bestpath_aspath_multipath_relax: 'true',
+    bestpath_compare_routerid:       'true',
+    bestpath_cost_community_ignore:  'true',
+    bestpath_med_confed:             'true',
+    bestpath_med_missing_as_worst:   'true',
+    bestpath_med_non_deterministic:  'true',
+  },
+}
+
+tests[:non_def_3] = {
+  desc:           '2.3 Non Defaults Part 3',
+  title_pattern:  "#{asn} default",
+  manifest_props: {
+    cluster_id:                             '10.0.0.1',
+    confederation_id:                       '99',
+    confederation_peers:                    '55 23.4 88 200.1',
+    enforce_first_as:                       'true',
+    fast_external_fallover:                 'true',
+    flush_routes:                           'false',
+    graceful_restart:                       'true',
+    graceful_restart_helper:                'true',
+    graceful_restart_timers_restart:        '130',
+    graceful_restart_timers_stalepath_time: '310',
+  },
+}
+
+tests[:non_def_4] = {
```

Comment:
What's the benefit or reasoning behind having four separate test cases rather than a single test case (or for that matter one test case per property or property set)? Do we have guidelines for this?